oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/DocumentStoreIndexerBase.java

```diff
@@ -207,6 +211,30 @@ private List<FlatFileStore> buildFlatFileStoreList(NodeState checkpointedState,
         return storeList;
     }
 
+    public IncrementalStore buildIncrementalStore(String initialCheckpoint, String finalCheckpoint) throws IOException, CommitFailedException {
+        IncrementalStoreBuilder builder;
+        IncrementalStore incrementalStore;
+        Set<String> preferredPathElements = new HashSet<>();
+        Set<IndexDefinition> indexDefinitions = getIndexDefinitions();
+        for (IndexDefinition indexDf : indexDefinitions) {
+            preferredPathElements.addAll(indexDf.getRelativeNodeNames());
+        }
```

Comment:
you can use java streams and return an immutable set here
```suggestion
        Set<IndexDefinition> indexDefinitions = getIndexDefinitions();

        Set<String> preferredPathElements = indexDefinitions.stream()
            .flatMap(indexDf -> indexDf.getRelativeNodeNames().stream())
            .collect(Collectors.toUnmodifiableSet());
```